internal/controller/services/monitoring/monitoring_controller_support.go

```diff
@@ -97,14 +98,26 @@ func getTemplateData(ctx context.Context, rr *odhtypes.ReconciliationRequest) (m
 			templateData["StorageRetention"] = "90d"
 		}
 
-		// only when either storage or resources is set, we take replicas into account
-		// - if user did not set it / zero-value "0", we use default value of 2
-		// - if user set it to Y, we pass Y to template
-		var replicas int32 = 2 // default value to match monitoringstack CRD's default
-		if (metrics.Storage != nil || metrics.Resources != nil) && metrics.Replicas != 0 {
-			replicas = metrics.Replicas
+		// - if user explicitly set replicas, use their value
+		// - if metrics is configured (storage or resources) but no explicit replicas, use SNO-aware defaults
+		// - if no metrics config but SNO, enforce 1 replica
+		// - otherwise, rely on MonitoringStack CRD defaults
+		allowedByConfig := metrics.Storage != nil || metrics.Resources != nil
+		isSNO := isSingleNodeCluster(ctx, rr)
+
+		switch {
+		case metrics.Replicas != 0 && allowedByConfig:
+			templateData["Replicas"] = strconv.Itoa(int(metrics.Replicas))
+		case allowedByConfig:
+			if isSNO {
+				templateData["Replicas"] = "1"
+			} else {
+				templateData["Replicas"] = "2"
+			}
+		case isSNO:
+			templateData["Replicas"] = "1"
```

Comment:
Why is this case needed? If `allowedByConfig==false`, then metrics is not configured, which means MonitoringStack is not deployed, so we can omit this completely. Action to deploy MonitoringStack (deployMonitoringStack) will skip processing the templates:
https://github.com/opendatahub-io/opendatahub-operator/blob/67a3eab6ad256057511e8bcd354d1e15ddaa2304/internal/controller/services/monitoring/monitoring_controller_actions.go#L114-L121

Monitoring CR will either have metrics nil or one of the storage/resource non-nil.